backend/resolver/k8s/k8s.go

```diff
@@ -174,36 +174,66 @@ func (r *res) Search(ctx context.Context, typeURL, query string, limit uint32) (
 	switch typeURL {
 	case typeURLPod:
 		if idPattern.MatchString(query) {
-			for _, name := range clientsets {
+			mappedValues, err := meta.PatternValueMapping(&k8sv1api.Pod{}, query)
+			if err == nil && len(mappedValues["cluster"]) > 0 && len(mappedValues["namespace"]) > 0 && len(mappedValues["name"]) > 0 {
 				handler.Add(1)
-				go func(name string) {
+				go func(cluster, namespace, name string) {
 					defer handler.Done()
-					pod, err := r.svc.DescribePod(ctx, name, "", metav1.NamespaceAll, query)
+					pod, err := r.svc.DescribePod(ctx, cluster, cluster, namespace, name)
```

Comment:
double check the `clientset` here, I dont want to fanout again here if we have the cluster information but this relies on the `clientset` == `cluster`.